lib/streamlit/elements/selectbox.py

```diff
@@ -30,8 +30,47 @@
 from .form import current_form_id
 from .utils import check_callback_rules, check_session_state_rules
 
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+T = TypeVar("T")
+
 
 class SelectboxMixin:
+    @overload
+    def selectbox(
+        self,
+        label: str,
+        options: Sequence[T],
+        index: int = 0,
+        format_func: Callable[[Any], Any] = str,
+        key: Optional[Key] = None,
+        help: Optional[str] = None,
+        on_change: Optional[WidgetCallback] = None,
+        args: Optional[WidgetArgs] = None,
+        kwargs: Optional[WidgetKwargs] = None,
+        *,  # keyword-only arguments:
+        disabled: bool = False,
+    ) -> Optional[T]:
+        ...
+
+    @overload
+    def selectbox(
+        self,
+        label: str,
+        options: OptionSequence,
+        index: int = 0,
+        format_func: Callable[[Any], Any] = str,
+        key: Optional[Key] = None,
+        help: Optional[str] = None,
+        on_change: Optional[WidgetCallback] = None,
+        args: Optional[WidgetArgs] = None,
+        kwargs: Optional[WidgetKwargs] = None,
+        *,  # keyword-only arguments:
+        disabled: bool = False,
+    ) -> Optional[Any]:
+        ...
```

Comment:
I'm guessing the difference between the two overloads is to allow the type checker to be able to deduce the return type of a `selectbox` in the case where `options` is a regular `Sequence` vs one of the `OptionsSequence` types that can't be parametrized? Would recommend adding a clarifying comment about this so that it's a bit more obvious.

Reply:
Additionally, I think with these overloads, it's now possible to remove `Sequence[Any]` from the `OptionsSequence` type as that case should be captured by the first case.

Reply:
> I'm guessing the difference between the two overloads is to allow the type checker to be able to deduce the return type of a selectbox in the case where options is a regular Sequence vs one of the OptionsSequence types that can't be parametrized? Would recommend adding a clarifying comment about this so that it's a bit more obvious.

Your understanding is on point. Will add a comment.

Reply:
I took a few steps back, and I think I managed to gracefully turn OptionSequence into a generic type in such a way that the overload isn't needed anymore. I think this is a cleaner solution.